pkg/syncer/storagepool/storageclasswatch.go

```diff
@@ -325,7 +327,12 @@ func (w *StorageClassWatch) getDatastoreToPolicyCompatibility(ctx context.Contex
 	for _, policyID := range w.policyIds {
 		compat, err := w.vc.PbmCheckCompatibility(ctx, datastoreMorList, policyID)
 		if err != nil {
-			log.Errorf("Failed to check PBM compatibility: %s %s", datastoreMorList, policyID)
+			if soap.IsSoapFault(err) && strings.Contains(err.Error(), "Failed to fetch content for the provided profiles.") {
```

Comment:
where is this error message coming from - "Failed to fetch content for the provided profiles."?
if this is from vCenter, then we should take care of localization. We generally do not compare error messages and always rely on faults.

Reply:
+1. We should never check on error messages as there is possibility that this error messages might change in future.

Reply:
I agree with you in general. However, in this case the error returned by vcenter (pbm in this case) shows up as a generic SoapFault error. So had to look into the message string to identify the specific error I am trying to handle here.

Reply:
Can we request spbm team to add a specific fault for our need and add a TODO here to revisit this later?

Reply:
SPBM returns an InvalidArgument error already. It is just that the govmomi wraps it into a SoapFault.

When I first went looking how to unwrap this SoapFault, I did not see it. Now realized that govmomi already has methods to look into the underlying fault thrown from the vim vmodl. I have now updated the diff with that change and also manually tested that this works as expected. Thanks for prodding me to look again.